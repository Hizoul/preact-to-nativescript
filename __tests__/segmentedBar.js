import * as preactToNativeScript from "../index"

const h = preactToNativeScript.Preact.h

class SegmentedBar extends preactToNativeScript.Preact.Component {
  render() {
    const show = this.state ? this.state.show : false
    const segmentChildren = [h("segmentedBarItem", {title: "bla"})]
    segmentChildren.push(h("segmentedBarItem", {title: "asdf"}))
    if (show) {
      segmentChildren.push(h("segmentedBarItem", {title: "hasdf"}))
    }
    segmentChildren.push(h("segmentedBarItem", {title: "uztriurtzui"}))
    if (show) {
      segmentChildren.push(h("segmentedBarItem", {title: "mbvcm"}))
    }
    segmentChildren.push(h("segmentedBarItem", {title: "97634"}))
    if (show) {
      segmentChildren.push(h("segmentedBarItem", {title: "123465"}))
    }
    return h("page", {updateState: this.setState}, [
      h("stackLayout", null, [
        h("segmentedBar", null, segmentChildren)
      ])
    ])
  }
}

test("SegmentedBar", () => {
  let vnode = h(SegmentedBar)
  let resultingRender = preactToNativeScript.render(vnode)
  expect(resultingRender).toMatchSnapshot("Render of SegmentedBar")
  resultingRender._component.setState({show: true})
  resultingRender = preactToNativeScript.render(vnode, resultingRender.parentNode, resultingRender)
  expect(resultingRender).toMatchSnapshot("Render of changed SegmentedBar")
})
