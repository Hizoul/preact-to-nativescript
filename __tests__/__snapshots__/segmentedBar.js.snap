// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedBar: Render of SegmentedBar 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {
    "updateState": [Function],
  },
  "_component": SegmentedBar {
    "__b": null,
    "__c": null,
    "__d": false,
    "__k": undefined,
    "__p": null,
    "__r": undefined,
    "__s": null,
    "__x": false,
    "base": [Circular],
    "context": Object {},
    "props": Object {
      "children": Array [],
    },
    "state": Object {},
  },
  "_componentConstructor": [Function],
  "attributes": Array [],
  "childNodes": Array [
    StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "attributes": Array [],
      "childNodes": Array [
        SegmentedBar {
          "__n": "segmentedBar",
          "__preactattr_": Object {},
          "attributes": Array [],
          "childNodes": Array [
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "bla",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "bla",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "asdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "asdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "uztriurtzui",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "uztriurtzui",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "97634",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "97634",
            },
          ],
          "firstLoad": true,
          "items": Array [
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "bla",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "bla",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "asdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "asdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "uztriurtzui",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "uztriurtzui",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "97634",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "97634",
            },
          ],
          "loaded": false,
          "nodeName": "SEGMENTEDBAR",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  ],
  "content": StackLayout {
    "__n": "stackLayout",
    "__preactattr_": Object {},
    "attributes": Array [],
    "childNodes": Array [
      SegmentedBar {
        "__n": "segmentedBar",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "bla",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "bla",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "asdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "asdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "uztriurtzui",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "uztriurtzui",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "97634",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "97634",
          },
        ],
        "firstLoad": true,
        "items": Array [
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "bla",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "bla",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "asdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "asdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "uztriurtzui",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "uztriurtzui",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "97634",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "97634",
          },
        ],
        "loaded": false,
        "nodeName": "SEGMENTEDBAR",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "firstLoad": true,
    "loaded": false,
    "nodeName": "STACKLAYOUT",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
  },
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": [Function],
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`SegmentedBar: Render of changed SegmentedBar 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {
    "updateState": [Function],
  },
  "_component": SegmentedBar {
    "__b": null,
    "__c": null,
    "__d": false,
    "__k": undefined,
    "__p": null,
    "__r": undefined,
    "__s": null,
    "__x": false,
    "base": [Circular],
    "context": Object {},
    "props": Object {
      "children": Array [],
    },
    "state": Object {
      "show": true,
    },
  },
  "_componentConstructor": [Function],
  "attributes": Array [],
  "childNodes": Array [
    StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "attributes": Array [],
      "childNodes": Array [
        SegmentedBar {
          "__n": "segmentedBar",
          "__preactattr_": Object {},
          "attributes": Array [],
          "childNodes": Array [
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "bla",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "bla",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "asdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "asdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "hasdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "hasdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "uztriurtzui",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "uztriurtzui",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "mbvcm",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "mbvcm",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "97634",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "97634",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "123465",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "123465",
            },
          ],
          "firstLoad": true,
          "items": Array [
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "bla",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "bla",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "asdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "asdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "hasdf",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "hasdf",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "uztriurtzui",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "uztriurtzui",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "mbvcm",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "mbvcm",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "97634",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "97634",
            },
            SegmentedBarItem {
              "__n": "segmentedBarItem",
              "__preactattr_": Object {
                "title": "123465",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "SEGMENTEDBARITEM",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "title": "123465",
            },
          ],
          "loaded": false,
          "nodeName": "SEGMENTEDBAR",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  ],
  "content": StackLayout {
    "__n": "stackLayout",
    "__preactattr_": Object {},
    "attributes": Array [],
    "childNodes": Array [
      SegmentedBar {
        "__n": "segmentedBar",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "bla",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "bla",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "asdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "asdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "hasdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "hasdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "uztriurtzui",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "uztriurtzui",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "mbvcm",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "mbvcm",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "97634",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "97634",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "123465",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "123465",
          },
        ],
        "firstLoad": true,
        "items": Array [
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "bla",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "bla",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "asdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "asdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "hasdf",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "hasdf",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "uztriurtzui",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "uztriurtzui",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "mbvcm",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "mbvcm",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "97634",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "97634",
          },
          SegmentedBarItem {
            "__n": "segmentedBarItem",
            "__preactattr_": Object {
              "title": "123465",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "SEGMENTEDBARITEM",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "title": "123465",
          },
        ],
        "loaded": false,
        "nodeName": "SEGMENTEDBAR",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "firstLoad": true,
    "loaded": false,
    "nodeName": "STACKLAYOUT",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
  },
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": [Circular],
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": [Function],
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;
